### PR TITLE
JP-3327: Modify mirmrsptcorr data model

### PR DESCRIPTION
**Pull request by drlaw1558**
Implements the data model changes described by https://jira.stsci.edu/browse/CRDS-721

**Checklist**
- [x] added entry in `CHANGES.rst` (either in `Bug Fixes` or `Changes to API`)
- [ ] updated relevant tests
- [ ] updated relevant documentation
- [ ] updated relevant milestone(s)
- [x] added relevant label(s)

### CODE REVIEW

File: src/stdatamodels/jwst/datamodels/schemas/miri_mrsptcorr.schema.yaml [hunk 1]
```diff
@@ -9,27 +9,16 @@ allOf:
...
-- $ref: dq_def.schema
-- type: object
-  properties:
+    leakcor_table:
+      title: IFU spectral leak correction
+      fits_hdu: LEAKCOR
+      datatype:
+      - name: wavelength
+        datatype: float32
+      - name: frac_leak
```
Comment (hbushouse) [inline src/stdatamodels/jwst/datamodels/schemas/miri_mrsptcorr.schema.yaml:18]: Not a big deal, but unlike FITS keyword names, we don't need to be stingy with our table column names and hence you could (if you want) spell these out more completely, for readability. Same for `err_leak`.

**Review by hbushouse — APPROVED** [review]
Looks OK to me. Just one minor comment, which you're free to ignore if you wish.

File: src/stdatamodels/jwst/datamodels/schemas/miri_mrsptcorr.schema.yaml (hunk 1 above)
Comment (drlaw1558) [reply src/stdatamodels/jwst/datamodels/schemas/miri_mrsptcorr.schema.yaml:18]: I'd be inclined to leave it as-is, for consistency with a draft file that's been in offline use via Box for a couple weeks

Comment (hbushouse) [reply src/stdatamodels/jwst/datamodels/schemas/miri_mrsptcorr.schema.yaml:18]: That's fine. Like I said, it's not a big deal.

**Comment by hbushouse** [comment]
Started a regtest run, pointing to this PR commit, at https://plwishmaster.stsci.edu:8081/job/RT/job/JWST-Developers-Pull-Requests/905

**Comment by hbushouse** [comment]
Regression test run was clean. I'm merging.

**Pull request merged**